src/Grpc.Net.Client/Balancer/Internal/ChildHandlerLoadBalancer.cs

```diff
@@ -0,0 +1,248 @@
+﻿#region Copyright notice and license
+
+// Copyright 2019 The gRPC Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Grpc.Net.Client.Configuration;
+using Grpc.Shared;
+
+namespace Grpc.Net.Client.Balancer.Internal
+{
+    /// <summary>
+    /// The load balancer is responsible for creating the real load balancer, and changing it if the resolver
+    /// returns a service config with a new policy name.
+    /// </summary>
+    internal sealed class ChildHandlerLoadBalancer : LoadBalancer
+    {
+        private static readonly PickFirstConfig DefaultConfig = new PickFirstConfig();
+        private static readonly BalancerAttributesKey<LoadBalancer> LoadBalancerKey = new BalancerAttributesKey<LoadBalancer>("LoadBalancer");
+
+        private readonly IChannelControlHelper _controller;
+        private readonly ServiceConfig? _initialServiceConfig;
+        private readonly ConnectionManager _connectionManager;
+        private readonly object _lock = new object();
+
+        internal (LoadBalancer LoadBalancer, string Name)? _current;
```

Comment:
Can you elaborate a bit on why there is a _current and _pending load balancer? Eventually I saw that this is updated when the state is updated by the IChannelControlHelper but it's not immediately obvious.

Reply:
Done